packages/feature-layer/src/featureServiceHelpers.ts

```diff
@@ -680,6 +705,52 @@ export function _getFieldVisibilityUpdates(fieldInfo: any): any[] {
   return visibilityUpdates;
 }
 
+/**
+ *  view field domains can contain different values than the source feature service field domains
+ *  use the cached domain when it differs from the source view field domain
+ *
+ * @param fieldInfo current view layer or table fieldInfo
+ * @param fieldUpdates any existing field updates
+ * @return Array of fields to be updated
+ * @protected
+ */
+export function _validateDomains(fieldInfo: any, fieldUpdates: any[]) {
+  const domainFields: any[] = [];
+  const domainNames: string[] = [];
+
+  // loop through the cached fields from the source view we are cloning
+  fieldInfo.sourceFields.forEach((field: any) => {
+    if (field.hasOwnProperty("domain") && field.domain) {
+      domainFields.push(field.domain);
+      domainNames.push(String(field.name).toLocaleLowerCase());
+    }
+  });
+
+  // loop through the fields from the new view service
+  // add an update when the domains don't match
+  fieldInfo.newFields.forEach((field: any) => {
+    const i: number = domainNames.indexOf(
+      String(field.name).toLocaleLowerCase()
+    );
+    if (i > -1 && field.hasOwnProperty("domain") && field.domain) {
+      if (JSON.stringify(field.domain) !== JSON.stringify(domainFields[i])) {
+        // should mixin the update if the field already has some other update
+        let hasUpdate: boolean = false;
+        fieldUpdates.forEach((update: any) => {
```

Comment:
How about using [`some`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some) instead of `forEach`? That way, you can break out of the loop after the match is found.

Reply:
@MikeTschudi I'd not used that before but I like it...thanks! Will switch it up

Reply:
@MikeTschudi actually I still need to alter the array by setting the domain property on the instance where the name matches. I could still use a standard for loop and break out when the match is found though.

Reply:
The change would be to
```
fieldUpdates.some((update: any) => {
    if (update.name === field.name) {
        hasUpdate = true;
        update.domain = domainFields[i];
    }
    return hasUpdate;
});
```
Wouldn't this accomplish the domain-setting for each matching newField?

Reply:
@MikeTschudi ahh..you are correct...could do it that way

Reply:
Up to you to change or not; please merge when you want